docs/guidelines/platforms/android/guideline_understandable_android.md

```diff
@@ -140,6 +181,41 @@ TextField(
 
 - Not providing enough context for the views / composables that expect user interaction.
 
+### Error Suggestion (WCAG 3.3.3 - Level AA)
+
+To provide the best user experience, the application should suggest solutions for input errors when they are detected, unless doing so compromises security or the content's purpose.
+
+> This technique covers point *3.3.3 Error Suggestion - Level AA of the WCAG standard.*
+
+#### ✅ Success technique(s)
+
+When an error is detected, the application should provide the user with a suggestion on how to correct it. This can be done by providing a hint or a description of the error, along with suggested correction text to guide the user.
+
+Using an alert dialog is recommended because it ensures that users see the alert content first, making them aware of any errors. Without this notification, users may not realise an error has occurred and might mistakenly believe the form is not functioning correctly.
+
+Regardless of the method used, the error message should indicate the location of the error and provide a clear explanation of the issue.
+
+### Error Prevention (Legal, Financial, Data) (WCAG 3.3.4 - Level AA)
+
+To prevent users from making mistakes, the application should provide a mechanism before finalizing a transaction that involves legal, financial, or data-sensitive actions.
+
+> This technique covers point *3.3.4 Error Prevention (Legal, Financial, Data) - Level AA of the WCAG standard.*
+
+#### ✅ Success technique(s)
+
+To satisfy this criterion, the app should provide at least one of the following mechanisms:
+- **Reversible**: The submission is reversible.
+- **Checked**: Data entered by the user is checked for input errors and the user is provided with an opportunity to correct them.
+- **Confirmed**: A mechanism is available for reviewing, confirming, and correcting information before finalizing the submission.
```

Comment:
I am having a bit of trouble understanding how this would look, do you have any concrete example? 🤔

Reply:
@zavrskina I've added examples for each mechanism 😄